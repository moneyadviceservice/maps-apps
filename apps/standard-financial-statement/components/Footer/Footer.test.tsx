import { render, screen } from '@testing-library/react';

import '@testing-library/jest-dom';
import { Footer } from '.';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {},
  }),
}));

describe('Footer component', () => {
  it('renders correctly', () => {
    render(
      <Footer
        assetPath=""
        footerPath={''}
        footerLogo={{
          image: {
            _path: '',
            width: 0,
            height: 0,
            mimeType: '',
          },
          altText: '',
        }}
        footerLinks={[
          {
            linkTo: '/accessibility',
            text: 'Accessibility',
            description: null,
          },
          { linkTo: '/cookies', text: 'Cookies', description: null },
          { linkTo: '/privacy', text: 'Privacy', description: null },
          { linkTo: '/sitemap', text: 'Sitemap', description: null },
        ]}
      />,
    );
    const footer = screen.getByTestId('footer');
    expect(footer).toMatchSnapshot();
  });
});
