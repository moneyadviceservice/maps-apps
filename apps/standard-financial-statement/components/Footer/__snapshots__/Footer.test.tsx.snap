// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly 1`] = `
<footer
  class="relative z-1"
  data-testid="footer"
>
  <div
    class="flex flex-col py-8 bg-gray-500 md:flex-row md:py-6"
  >
    <div
      class="container-auto flex flex-col text-white md:flex-row max-w-[1272px]"
    >
      <ul
        class="relative flex flex-row flex-wrap mt-8 md:ml-auto md:mt-0"
      >
        <li
          class="text-white last:mr-0 relative px-2.5 sm:px-4 first:pl-0 flex items-center"
        >
          <a
            class="inline-flex items-center gap-2 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none text-sm text-gray-100 visited:text-gray-100 after:content-[""] after:absolute after:h-[16px] after:w-[1px] after:bg-blue-300 after:-m-[8px] after:top-[50%] after:right-[8px] hover:text-green-300 hover:underline"
            href="/en/accessibility"
          >
            Accessibility
          </a>
        </li>
        <li
          class="text-white last:mr-0 relative px-2.5 sm:px-4 first:pl-0 flex items-center"
        >
          <a
            class="inline-flex items-center gap-2 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none text-sm text-gray-100 visited:text-gray-100 after:content-[""] after:absolute after:h-[16px] after:w-[1px] after:bg-blue-300 after:-m-[8px] after:top-[50%] after:right-[8px] hover:text-green-300 hover:underline"
            href="/en/cookies"
          >
            Cookies
          </a>
        </li>
        <li
          class="text-white last:mr-0 relative px-2.5 sm:px-4 first:pl-0 flex items-center"
        >
          <a
            class="inline-flex items-center gap-2 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none text-sm text-gray-100 visited:text-gray-100 after:content-[""] after:absolute after:h-[16px] after:w-[1px] after:bg-blue-300 after:-m-[8px] after:top-[50%] after:right-[8px] hover:text-green-300 hover:underline"
            href="/en/privacy"
          >
            Privacy
          </a>
        </li>
        <li
          class="text-white last:mr-0 relative px-2.5 sm:px-4 first:pl-0 flex items-center"
        >
          <a
            class="inline-flex items-center gap-2 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none text-sm text-gray-100 visited:text-gray-100 after:content-[""] after:absolute after:h-[16px] after:w-[1px] after:bg-blue-300 after:-m-[8px] after:top-[50%] after:right-[8px] hover:text-green-300 hover:underline after:hidden"
            href="/en/sitemap"
          >
            Sitemap
          </a>
        </li>
      </ul>
    </div>
  </div>
</footer>
`;
